HanksMineralEmporium/Service/AuthenticationService/ AuthenticationService.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Web;
+using HanksMineralEmporium.Core.UserManagement;
+using HanksMineralEmporium.Core.UserManagement.Exception;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using MudBlazor;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Http;
+using HanksMineralEmporium.Service.AuthenticationService.Exception;
+/// <summary>
+/// Initializes a new instance of the <see cref="AuthenticationService"/> class.
+/// </summary>
+/// <param name="userManager">An implementation of <see cref="IUserManager"/> for user management.</param>
+/// <param name="serviceProvider">Service provider for dependency injection.</param>
+/// <remarks>
+/// The <paramref name="userManager"/> parameter should not be null.
+/// </remarks>
+/// <summary>
+/// Validates user credentials and attempts to log in the user.
+/// </summary>
+/// <param name="username">The username of the user.</param>
+/// <param name="password">The password of the user.</param>
+
+namespace HanksMineralEmporium.Service.AuthenticationService;
+
+public class  AuthenticationService : IAuthenticationService
+{
+    private IUserManager _userManager;
+    public AuthenticationService(IUserManager userManager)
+    {
+        _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+    }
+
+    public async Task RegisterUserAsync(string username, string password)
+    {
+        // Create the user and validate
+        var user = await _userManager.LoadUserAsync(username);
+        await RegisterUserAsync(user.Username,user.Password);
+    }
+
+    public async Task LoginUserAsync(string username, string password)
+    {
+        var user = await _userManager.LoadUserAsync(username);
+        // hash password
+        var HashPassword = Shared.Util.PasswordHashUtil.HashPassword(password);
+        if(user.Password.Equals(HashPassword))
+        {
+            StoreUserInSession(user);
+        }             
+        else{throw new PasswordMismatchException("The provided passwords do not match.");}
```

Comment:
I would change the error message to "Incorrect password" since this is what will be displayed to users of the website. Also please unwrap the else statement into:
else
{
    ...
}
This is to follow c# coding style conventions